lib/gui/components/update-notifier.js

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/**
+ * @module Etcher.Components.UpdateNotifier
+ */
+
+const angular = require('angular');
+const electron = require('electron');
+const Bluebird = require('bluebird');
+const _ = require('lodash');
+const settings = require('../models/settings');
+const units = require('../../shared/units');
+
+const MODULE_NAME = 'Etcher.Components.UpdateNotifier';
+const UpdateNotifier = angular.module(MODULE_NAME, [
+  require('../os/open-external/open-external'),
+  require('../modules/analytics')
+]);
+
+UpdateNotifier.service('UpdateNotifierService', function(AnalyticsService, OSOpenExternalService) {
+
+  /**
+   * @summary The number of days the update notifier can be put to sleep
+   * @constant
+   * @private
+   * @type {Number}
+   */
+  this.UPDATE_NOTIFIER_SLEEP_DAYS = 7;
+
+  /**
+   * @summary Determine if it's time to check for updates
+   * @function
+   * @public
+   *
+   * @param {Object} [options] - options
+   * @param {Boolean} [options.ignoreSleepUpdateCheck] - ignore sleep update check
+   * @param {Number} [options.lastSleptUpdateNotify] - last slept update notify time
+   * @returns {Boolean} should check for updates
+   *
+   * @example
+   * if (UpdateNotifierService.shouldCheckForUpdates({
+   *   ignoreSleepUpdateCheck: false,
+   *   lastSleptUpdateNotify: Date.now()
+   * })) {
+   *   console.log('We should check for updates!');
+   * }
+   */
+  this.shouldCheckForUpdates = (options = {}) => {
+    if (_.some([
+      !options.lastSleptUpdateNotify,
+      _.get(options, [ 'ignoreSleepUpdateCheck' ], false)
+    ])) {
+      return true;
+    }
+
+    if (options.lastSleptUpdateNotify - Date.now() > units.daysToMilliseconds(this.UPDATE_NOTIFIER_SLEEP_DAYS)) {
```

Comment:
So `options.lastSleptUpdateNotify - Date.now()` will always be negative because `options.lastSleptUpdateNotify` will be an older `Date.now()`, hence `options.lastSleptUpdateNotify <= Date.now()` is always true and this results in a negative number that decreases over time, instead of eventually growing past the seven days worth of seconds. So we should swap this to `Date.now() - options.lastSleptUpdateNotify > ...`. I think this was my bad for suggesting this is how it should be in the diff I made, sorry about that.